sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/BaseRecognizer.java

```diff
@@ -98,4 +106,58 @@ public void executeOrderBy(SQLOrderBy sqlOrderBy,SQLASTVisitor visitor) {
     public String getOriginalSQL() {
         return originalSQL;
     }
+
+    protected abstract SQLStatement getAst();
+
+    @Override
+    public boolean isSqlSyntaxSupports() {
+        SQLASTVisitor visitor = new SQLASTVisitorAdapter() {
+            @Override
+            public boolean visit(SQLJoinTableSource x) {
+                //just like: UPDATE table a INNER JOIN table b ON a.id = b.pid ...
+                throw new NotSupportYetException("not support the sql syntax with join table:" + x
+                        + "\nplease see the doc about SQL restrictions https://seata.io/zh-cn/docs/user/sqlreference/dml.html");
+            }
+
+            @Override
+            public boolean visit(SQLInSubQueryExpr x) {
+                //just like: ...where id in (select id from t)
+                throw new NotSupportYetException("not support the sql syntax with InSubQuery:" + x
+                        + "\nplease see the doc about SQL restrictions https://seata.io/zh-cn/docs/user/sqlreference/dml.html");
+            }
+
+            @Override
+            public boolean visit(SQLSubqueryTableSource x) {
```

Comment:
for update ?

Reply:
That's right. And the example statement is to illustrate the scope of SQLSubqueryTableSource